tuf/utils/x509.go

```diff
@@ -538,7 +538,7 @@ func CertBundleToKey(leafCert *x509.Certificate, intCerts []*x509.Certificate) (
 }
 
 // NewCertificate returns an X509 Certificate following a template, given a GUN and validity interval.
-func NewCertificate(gun string, startTime, endTime time.Time) (*x509.Certificate, error) {
+func NewCertificate(gun data.GUN, startTime, endTime time.Time) (*x509.Certificate, error) {
```

Comment:
Should leave this as type string as it's not always a gun. I've filed an issue to fix the naming of the argument (https://github.com/docker/notary/issues/1093)